### ISSUE TITLE
Speed up CI with "official" docker caching (available with github actions)

### ISSUE THREAD

**Issue by hpvd** [apache/pulsar#12519]
**Is your enhancement request related to a problem? Please describe.**
CI could always be faster...
great speedups were already introduced with:
https://github.com/apache/pulsar/pull/10001

**Describe the solution you'd like**
I'm not sure but maybe looking into some infos/details documented here (directly and in linked articles) may help to speed up further:

> 
> 
> Hi, I wrote [the article](https://dev.to/dtinth/caching-docker-builds-in-github-actions-which-approach-is-the-fastest-a-research-18ei) mentioned above which has been outdated and [superseded by another Evil Martians’ article](https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching).
> 
> TL;DR — Docker has published an official action [docker/build-push-action](https://github.com/marketplace/actions/build-and-push-docker-images) which [has an option to make Docker use GitHub Cache API directly as a cache storage](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#github-cache). With this settings my build time dropped from 5 minutes to 1 minute.
> 
> ```yaml
>           cache-from: type=gha
>           cache-to: type=gha,mode=max
> ```

source: comment of @dtinth, on Oct 27, 2021
in github/actions/cache/docker caching
https://github.com/actions/cache/issues/31#issuecomment-952545077

**Comment by tisonkun** [comment]
cc @lhotari is this helpful for our current CI workflow?

**Comment by lhotari** [comment]
> cc @lhotari is this helpful for our current CI workflow?

@tisonkun No.

**Comment by lhotari** [comment]
Closing since docker images are cached in the current workflow.
